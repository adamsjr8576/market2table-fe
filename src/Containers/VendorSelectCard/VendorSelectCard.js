import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './VendorSelectCard.scss';
import { addSelectedVendor } from '../../actions';

export const VendorSelectCard = ({ name, vendor }) => {

  const dispatch = useDispatch();

  return (
    <article className='vendor-select-article'>
      <h2 className='vendor-article-header'>{name}</h2>
      <section className='vendor-article-button-container'>
        <button className='vendor-article-markets-button'>Link to Markets</button>
        <Link to='/vendor/account/form' className='link-edit-vendor-form'>
          <button className='vendor-article-edit-button' onClick={() => dispatch(addSelectedVendor(vendor))}>Edit</button>
        </Link>
      </section>
    </article>
  )
}

export default VendorSelectCard;
