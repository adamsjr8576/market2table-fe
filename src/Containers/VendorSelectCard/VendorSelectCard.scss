.vendor-select-article {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  border-bottom: 3px solid #3B3D39;
}

.vendor-article-header {
  margin-left: 25px;
  min-width: 125px;
}

.vendor-article-button-container {
  width: 35%;
  min-width: 200px;
  display: flex;
  align-items: center;
  justify-content: space-around;
  margin-right: 20px;
}

.vendor-article-markets-button {
  width: 50%;
  padding: 5px;
  font-size: 1em;
  border: 2px solid #3B3D39;
  background-color: #F0F0EF;
  outline: 0;
  border-radius: 10px;
  cursor: pointer;
}

.vendor-article-markets-button:hover {
  background-color: #C4D1B2;
  border: 2px solid #5E7838;
}

.vendor-article-edit-button {
  width: 25%;
  padding: 5px;
  font-size: 1em;
  border: 2px solid #3B3D39;
  background-color: #F0F0EF;
  outline: 0;
  border-radius: 10px;
  cursor: pointer;
}

.vendor-article-edit-button:hover {
  background-color: #C4D1B2;
  border: 2px solid #5E7838;
}
