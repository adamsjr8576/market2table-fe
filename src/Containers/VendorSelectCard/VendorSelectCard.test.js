import React from 'react';
import { shallow } from 'enzyme';
import VendorSelectCard from './VendorSelectCard';
import { useDispatch } from 'react-redux';


jest.mock("react-redux", () => ({
    useDispatch: () => jest.fn()
}));

describe('VendorSelectCard', () => {
    let mockAddSelectedVendor = jest.fn();
    let mockId = 2;
    let mockName = 'foster';
    let mockVendor = 'fosters booth';

    it('should match the VendorSelectCard snapshot', () => {
        let wrapper = shallow(<VendorSelectCard
            id={mockId}
            name={mockName}
            vendor={mockVendor}
            addSelectedVendor={mockAddSelectedVendor}
          />);
        expect(wrapper).toMatchSnapshot();
    })

    it('should invoke dispatch on article markets button', () => {
        let wrapper = shallow(<VendorSelectCard
            id={mockId}
            name={mockName}
            vendor={mockVendor}
            addSelectedVendor={mockAddSelectedVendor}
          />);

      wrapper.find('.vendor-article-markets-button').simulate('click');
      expect(mockAddSelectedVendor).toHaveBeenCalled();
    })

});
