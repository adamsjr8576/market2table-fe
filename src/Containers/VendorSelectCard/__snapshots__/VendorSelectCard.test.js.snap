// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorSelectCard should invoke dispatch on article markets button 1`] = `
<article
  className="vendor-select-article"
>
  <h2
    className="vendor-article-header"
  >
    foster
  </h2>
  <section
    className="vendor-article-button-container"
  >
    <Link
      className="link-market-vendor-link"
      to="/vendor/account/link/2"
    >
      <button
        className="vendor-article-markets-button"
        onClick={[Function]}
      >
        Link to Markets
      </button>
    </Link>
    <Link
      className="link-edit-vendor-form"
      to="/vendor/account/form"
    >
      <button
        className="vendor-article-edit-button"
        onClick={[Function]}
      >
        Edit
      </button>
    </Link>
  </section>
</article>
`;

exports[`VendorSelectCard should match the VendorSelectCard snapshot 1`] = `
<article
  className="vendor-select-article"
>
  <h2
    className="vendor-article-header"
  >
    foster
  </h2>
  <section
    className="vendor-article-button-container"
  >
    <Link
      className="link-market-vendor-link"
      to="/vendor/account/link/2"
    >
      <button
        className="vendor-article-markets-button"
        onClick={[Function]}
      >
        Link to Markets
      </button>
    </Link>
    <Link
      className="link-edit-vendor-form"
      to="/vendor/account/form"
    >
      <button
        className="vendor-article-edit-button"
        onClick={[Function]}
      >
        Edit
      </button>
    </Link>
  </section>
</article>
`;
